### PR TITLE
Upgrade operator to v0.78.0

### PR DESCRIPTION
**Pull request by bryan-aguilar**
Upgrades operator to v0.78.0

### CODE REVIEW

**Review by TylerHelmuth — COMMENTED** [review]
@bryan-aguilar can you try removing the default go image values from the values.yaml?  I had to include those in 0.77.0 because of https://github.com/open-telemetry/opentelemetry-operator/issues/1756 but they shouldn't be needed anymore

**Comment by bryan-aguilar** [comment]
@TylerHelmuth done. I also bumped to `kube-rbac-proxy` version to `v0.14.2`.  Release notes [here](https://github.com/brancz/kube-rbac-proxy/releases).

**Comment by bryan-aguilar** [comment]
Is there an authentication issue with submitting fork origin pull requests? 
`
  logger.go:42: 18:02:24 | prometheus-config-validation | 2023-06-05 17:59:55 +0000 UTC	Warning	Pod promreceiver-allocatorconfig-targetallocator-6b5c49c68d-qm6ns.spec.containers{ta-container}		Failed	Failed to pull image "local/opentelemetry-operator-targetallocator:e2e": rpc error: code = Unknown desc = failed to pull and unpack image "docker.io/local/opentelemetry-operator-targetallocator:e2e": failed to resolve reference "docker.io/local/opentelemetry-operator-targetallocator:e2e": pull access denied, repository does not exist or may require authorization: server message: insufficient_scope: authorization failed
`

**Comment by TylerHelmuth** [comment]
@bryan-aguilar since a local container is being referenced for those tests you will most likely need to do something in the CI like this: https://github.com/open-telemetry/opentelemetry-helm-charts/blob/28dc76de32b4477dbbaa500e3724b88a6280efa6/.github/workflows/operator-test.yaml#L43-L49

**Comment by TylerHelmuth** [comment]
Oh I think it is the same image.  You'll need to add more cases to that step in the workflow

File: .github/workflows/operator-test.yaml [hunk 1]
```diff
@@ -47,6 +47,7 @@ jobs:
 
           sed -i "s/$LOCAL_TARGET_ALLOCATOR_IMG/${PUBLIC_TARGET_ALLOCATOR_IMG}/g" ./opentelemetry-operator/tests/e2e/smoke-targetallocator/00-install.yaml
           sed -i "s/$LOCAL_TARGET_ALLOCATOR_IMG/${PUBLIC_TARGET_ALLOCATOR_IMG}/g" ./opentelemetry-operator/tests/e2e/targetallocator-features/00-install.yaml
+          sed -i "s/$LOCAL_TARGET_ALLOCATOR_IMG/${PUBLIC_TARGET_ALLOCATOR_IMG}/g" tests/e2e/prometheus-config-validation/*.yaml
```
Comment (TylerHelmuth) [inline .github/workflows/operator-test.yaml:50]:
```suggestion
          sed -i "s/$LOCAL_TARGET_ALLOCATOR_IMG/${PUBLIC_TARGET_ALLOCATOR_IMG}/g" ./opentelemetry-operator/tests/e2e/prometheus-config-validation/*.yaml
```

**Review by TylerHelmuth — APPROVED** [review]

**Pull request merged**